components/scream/src/physics/share/physics_test_data.cpp

```diff
@@ -0,0 +1,141 @@
+#include "physics_test_data.hpp"
+
+#include "ekat/ekat_assert.hpp"
+#include "ekat/kokkos/ekat_kokkos_utils.hpp"
+#include "ekat/ekat_pack_kokkos.hpp"
+
+#include <random>
+
+namespace scream {
+
+PhysicsTestData::PhysicsTestData(Int shcol_, const std::vector<Real**>& ptrs_c, const std::vector<Int**>& idx_c) :
+  PhysicsTestData(shcol_, 0, 0, {}, {}, ptrs_c, idx_c)
+{}
+
+PhysicsTestData::PhysicsTestData(Int shcol_, Int nlev_, const std::vector<Real**>& ptrs, const std::vector<Real**>& ptrs_c, const std::vector<Int**>& idx_c) :
+  PhysicsTestData(shcol_, nlev_, 0, ptrs, {}, ptrs_c, idx_c) {}
+
+PhysicsTestData::PhysicsTestData(Int shcol_, Int nlev_, Int nlevi_,
+                                 const std::vector<Real**>& ptrs, const std::vector<Real**>& ptrs_i,
+                                 const std::vector<Real**>& ptrs_c, const std::vector<Int**>& idx_c) :
+  shcol(shcol_),
+  nlev(nlev_),
+  nlevi(nlevi_),
+  m_total(shcol_ * nlev_),
+  m_totali(shcol_ * nlevi_),
+  m_ptrs(ptrs),
+  m_ptrs_i(ptrs_i),
+  m_ptrs_c(ptrs_c),
+  m_indices_c(idx_c),
+  m_data(m_ptrs.size() * m_total + m_ptrs_i.size() * m_totali + m_ptrs_c.size() * shcol, 0),
+  m_idx_data(idx_c.size() * shcol, 0)
+{
+  init_ptrs();
+}
+
+PhysicsTestData::PhysicsTestData(const PhysicsTestData &rhs,
+                                 const std::vector<Real**>& real_ptrs, const std::vector<Int**>& int_ptrs) :
+  shcol(rhs.shcol),
+  nlev(rhs.nlev),
+  nlevi(rhs.nlevi),
+  m_total(rhs.m_total),
+  m_totali(rhs.m_totali),
+  m_ptrs(rhs.m_ptrs.size()),
+  m_ptrs_i(rhs.m_ptrs_i.size()),
+  m_ptrs_c(rhs.m_ptrs_c.size()),
+  m_indices_c(rhs.m_indices_c.size()),
+  m_data(rhs.m_data),
+  m_idx_data(rhs.m_idx_data)
+{
+  const size_t expected_real_ptrs = m_ptrs.size() + m_ptrs_i.size() + m_ptrs_c.size();
+  EKAT_REQUIRE_MSG(real_ptrs.size() == expected_real_ptrs, real_ptrs.size() << " != " << expected_real_ptrs);
```

Comment:
I will improve the beginning of the message. We may have to agree to disagree on `<<`. I personally love when assert macros support streams in the message since it is quite tedious in C++ to convert things to strings.

Reply:
Do you think an EKAT macro like the following would be helpful? 
```
EKAT_REQUIRE_EQ (lhs, rhs, msg)
```
which checks lhs==rhs and prints msg if test fails (and then throws). And maybe similarly for <,<=,>,>=. That way we pass all the info needed withouth the need of streams. 

If so, I'll open a feature issue in EKAT.

Ps: it is quite handy when asserts support <<. My concern are that a) someone might think the return value of real_ptrs.size() supports <<, and b) if someone wraps the msg arg in the ekat macro in parentheses, we're in trouble. But these are just my fears. I won't argue with that being really handy.

Reply:
I think it's probably better to just allow stream operations and not have a big increase in the number of macros, but I don't care to much one way or the other.